docs/development.rst

```diff
@@ -354,3 +354,108 @@ You can deploy a hosted demo instance of Kuma by following these steps:
+    mysql> source ./scripts/create_read_only_user.sql
+    mysql> quit
+
+#. Create a ``.env`` file in the repository root directory, and add these
+   settings::
+
+    MAINTENANCE_MODE=True
+    DATABASE_USER=kuma_ro
+
+   Using a read-only database user is not required in maintenance mode. You can run
+   in maintenance mode just fine with only this setting::
+
+    MAINTENANCE_MODE=True
+
+   and going with a database user that has write privileges. The read-only database
+   user simply provides a level of safety as well as notification (for example, an
+   exception will be raised if an attempt to write the database slips through).
+
+#. If you were already running your local Docker instance, you need to bring it
+   down hard and start again::
+
+    docker-compose down
+    docker-compose up -d
+
+#. You may need to recompile your static assets and then restart::
+
+    docker-compose exec web make build-static
+    docker-compose restart web
+
+You should be good to go!
+
+There is a set of integration tests for maintenance mode. If you'd like to run
+them against your local Docker instance, you must first have done the
+following:
+
+#. Loaded the latest sample database (see :ref:`provision-the-database`).
+#. Ensured that the test document "en-US/docs/User:anonymous:uitest" has been
+   rendered (all of its macros have been executed). You can check this by
+   browsing to `http://localhost:8000/en-US/docs/User:anonymous:uitest`_. If
+   there is no message about un-rendered content, you are good to go. If there
+   is a message about un-rendered content, you will have to put your local
+   Docker instance back into non-maintenance mode, and render the document:
+
+   * ``docker-compose down``
+   * Configure your ``.env`` file for non-maintenance mode::
+
+       MAINTENANCE_MODE=False
+       DATABASE_USER=root
+
+   * ``docker-compose up -d``
+   * Using your browser, do a shift-reload on
+     `http://localhost:8000/en-US/docs/User:anonymous:uitest`_
+
+   and then put your local Docker instance back in maintenance mode:
+
+   * ``docker-compose down``
+   * Configure your ``.env`` file for maintenance mode::
+
+       MAINTENANCE_MODE=True
+       DATABASE_USER=kuma_ro
+
+   * ``docker-compose up -d``
+
+#. Configure your environment with DEBUG=False because the maintenance-mode
+   integration tests check for the non-debug version of the not-found page::
+
+       DEBUG=False
+       MAINTENANCE_MODE=True
+       DATABASE_USER=kuma_ro
+
+   This, in turn, will also require you to recompile your static assets::
+
+       docker-compose down
+       docker-compose up -d
+       docker-compose exec web ./manage.py compilejsi18n
+       docker-compose exec web ./manage.py collectstatic
+       docker-compose restart web
+
+Now you should be ready for a successful test run::
+
+    py.test -m "maintenance_mode and not search" tests/functional --base-url http://localhost:8000 --driver Chrome --driver-path /path/to/chromedriver
```

Comment:
My memory is that geckodriver and current Firefox requires Selenium 3, and we're using Selenium 2, here and maybe in Jenkins as well.  We should update, but not in this PR.  Maybe a contributor could give it a try...